### ISSUE TITLE
Reproducibility and future plans

### ISSUE THREAD

**Issue by clausmichele** [interTwin-eu/openeo-flood-mapper-local#7]
@ms801221 @fl0roth @MartinSchobben @SwamyDev
I was wondering what's the plan in terms of usage of this code.

From what I understand:
- Currently it's using pre-defined netCDFs with Sentinel-1 data (backscatter and local incidence angle) and a netCDF containing pre-computed parameters of the fitted function to the current area.
- It demonstrates how to use openEO to build and run the flood detection workflow.
- The plan for moving forward is to use the Sentinel-1 Sigma0 data and openEO back-end provided by EODC (instead of the local data and processing).

Some open points which are not documented yet are:
- How to compute the fitted parameters? Is it planned to provide the code for it?
- What are the specifications of the input data? (Resolution, other important parameters used for the pre-processing?) Please specify them if possible.

**Comment by MartinSchobben** [comment]
> * Currently it's using pre-defined netCDFs with Sentinel-1 data (backscatter and local incidence angle) and a netCDF containing pre-computed parameters of the fitted function to the current area.
> 
> * It demonstrates how to use openEO to build and run the flood detection workflow.
> 
> * The plan for moving forward is to use the Sentinel-1 Sigma0 data and openEO back-end provided by EODC (instead of the local data and processing).

That is correct but EODC faces some difficulties with the openEO server at the moment, so this will take some time to resolve. If that is resolved, we use the same workflow on the EODC backend, or, at least, try to stay as close as possible to the local processing pipeline.

> * How to compute the fitted parameters? Is it planned to provide the code for it?

Fitted parameters will be provided by EODC. At the moment it does not look like we will delineate the derivation in ~EODC~ openEO syntax.

> * What are the specifications of the input data? (Resolution, other important parameters used for the pre-processing?) Please specify them if possible.

This would be part of the EODC based workflow.

**Comment by clausmichele** [comment]
> 
> That is correct but EODC faces some difficulties with the openEO server at the moment, so this will take some time to resolve. If that is resolved, we use the same workflow on the EODC backend, or, at least, try to stay as close as possible to the local processing pipeline.

In the meanwhile, I see multiple possibilities to go forward:
1. Try to use a different openEO back-end/provider: Coperncius Data Space Ecosystem (CDSE) would have the data and the algorithms to produce Sentinel-1 Sigma0. With this data it would possible to check if the results using the computed data + local processing are comparable. If yes, it would be possible to try to run the full workflow there.
2. Try to use data from a public STAC Catalog like the one of Microsoft Planetary Computer: https://radiantearth.github.io/stac-browser/#/external/planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-1-rtc?.language=en  but from what I read here, this dataset contains Gamma0 and not Sigma0: https://forum.step.esa.int/t/s1-processing-product-vs-mspc-s1-rtc-product/40533

> 
> Fitted parameters will be provided by EODC. At the moment it does not look like we will delineate the derivation in EODC syntax.

There's no "EODC syntax", openEO process definitions are shared among many different back-ends. Anyway, to improve the reproducibility, even plain python code would be fine, it doesn't have to be expressed already in openEO. So there is no plan in sharing that part? I would see it as a big limitation.

> 
> This would be part of the EODC based workflow.

Again, the data pre-processing workflow should not be limited to EODC but can be defined in advance (you know what's your input data) and with which parameters is being pre-processed. Including a sample SNAP graph or referencing to a doc stating how to produce Sentinel-1 Sigma0 with it would be already a good starting point.

**Comment by MartinSchobben** [comment]
> 1. Try to use a different openEO back-end/provider: Coperncius Data Space Ecosystem (CDSE) would have the data and the algorithms to produce Sentinel-1 Sigma0. With this data it would possible to check if the results using the computed data + local processing are comparable. If yes, it would be possible to try to run the full workflow there.

Yes, that would be an idea, if there is 20 m Sentinel-1 sigma0, to check whether at least that part of the pipeline works.

> There's no "EODC syntax", openEO process definitions are shared among many different back-ends. Anyway, to improve the reproducibility, even plain python code would be fine, it doesn't have to be expressed already in openEO. So there is no plan in sharing that part? I would see it as a big limitation.

I meant "openEO" there. That is the decision for now; EODC will provide the harmonic parameters. 

> Again, the data pre-processing workflow should not be limited to EODC but can be defined in advance (you know what's your input data) and with which parameters is being pre-processed. Including a sample SNAP graph or referencing to a doc stating how to produce Sentinel-1 Sigma0 with it would be already a good starting point.

I see, that seems indeed useful.

**Comment by christophreimer** [comment]
@MartinSchobben all data needed to run the use case are available within the EODC backend.
As far as I'm aware of, we (EODC) haven't received any incident about something not working for the use case.
So lets have a meeting to discuss your issue directly.

**Comment by MartinSchobben** [comment]
@christophreimer I was of the understanding that a ticket was posted. But indeed let's discuss.

**Comment by clausmichele** [comment]
@christophreimer where are the model parameters stored? Are they exposed as an openEO collection or as STAC Collection?
Please include me as well when you'll organize a meeting.
